gridspeccer/core.py

```diff
@@ -146,7 +148,12 @@ def make_axes(gridspec, fig_kwargs=None):
     for k, gs_item in list(gridspec.items()):
         # we just add a label to make sure all axes are actually created
         log.debug("Creating subplot: %s", k)
-        axes[k] = fig.add_subplot(gs_item, label=k)
+        if isinstance(gs_item, gs.SubplotSpec):
+            axes[k] = fig.add_subplot(gs_item, label=k)
+        else:
```

Comment:
Let's not assume and explicitly enforce `SubplotSpec` or `(SubplotSpec, dict)`, throwing an error if not.

I would then simply create a `kwargs` dict that gets modified by whatever is set in the properties `dict`:

```python
kwargs = {"label": k}
gs_props = {}
if isinstance(gs_item, tuple):
    if len(gs_item) != 2 or not isinstance(gs_item[1], dict):
        raise ValueError("Subplots should be specified either as SubplotSpec or (SubplotSpec, dict).")
    gs_props = gs_item[1]
    gs_item = gs_item[0]

if gs_props.get('3d', False):
    kwargs['projection'] = '3d'

axes[k] = fig.add_subplot(gs_item, **kwargs)

Reply:
Did this, but moved the `kwargs` line 7 lines down, to reduce confusion with `gs_prop` (because I was confused at first)

Reply:
accept if you agree